sysutils/yabai/files/patch-makefile.diff

```diff
@@ -0,0 +1,34 @@
+--- makefile.orig	2021-01-22 17:41:47.000000000 -0800
++++ makefile	2021-01-22 17:55:53.000000000 -0800
+@@ -1,6 +1,6 @@
+ FRAMEWORK_PATH = -F/System/Library/PrivateFrameworks
+ FRAMEWORK      = -framework Carbon -framework Cocoa -framework CoreServices -framework SkyLight -framework ScriptingBridge
+-BUILD_FLAGS    = -std=c99 -Wall -g -O0 -fvisibility=hidden -mmacosx-version-min=10.13
++CFLAGS         += -std=c99 -Wall -DNDEBUG -O2 -fvisibility=hidden -mmacosx-version-min=10.13
+ BUILD_PATH     = ./bin
+ DOC_PATH       = ./doc
+ SCRIPT_PATH    = ./scripts
+@@ -16,13 +16,13 @@
+ 
+ all: clean-build $(BINS)
+ 
+-install: BUILD_FLAGS=-std=c99 -Wall -DNDEBUG -O2 -fvisibility=hidden -mmacosx-version-min=10.13
+ install: clean-build $(BINS)
++	cp $(BINS) $(DESTDIR)/$(PREFIX)/$(BINS)
+ 
+ $(OSAX_SRC): $(OSAX_PATH)/loader.m $(OSAX_PATH)/payload.m $(OSAX_PATH)/mach_bootstrap.c
+-	clang $(OSAX_PATH)/loader.m -shared -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/loader -framework Foundation
+-	clang $(OSAX_PATH)/payload.m -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/payload -framework Foundation -framework Carbon
+-	clang $(OSAX_PATH)/mach_bootstrap.c -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/mach_bootstrap -framework Carbon -lpthread
++	$(CC) $(OSAX_PATH)/loader.m -shared -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/loader -framework Foundation
++	$(CC) $(OSAX_PATH)/payload.m -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/payload -framework Foundation -framework Carbon
++	$(CC) $(OSAX_PATH)/mach_bootstrap.c -shared -fPIC -O2 -mmacosx-version-min=10.13 -o $(OSAX_PATH)/mach_bootstrap -framework Carbon -lpthread
```

Comment:
probably in these lines you should remove the same stuff as I mentioned above and I think probably use `CFLAGS` instead?

Reply:
I can definitely go through and re-write a good chunk of this makefile need be, but do you know why these flags in particular would be causing an issue? As far as I can see from the CI logs, it's selecting `CC=clang`, which indeed supports these flags, and seems to be erroring later on rather than here.

Reply:
Ah, scratch that -- seems as if the latest build is working! I haven't yet changed these lines (should I plumb through CFLAGS, even though the original Makefile does not use its equivalent BUILD_FLAGS here?), but it seems your suggestion for the overall CFLAGS changes indeed did the trick. Thanks for the help!

Reply:
according to the logs it is still adding `-shared -fPIC -O2 -mmacosx-version-min=10.13` these arguments to the compilation and not the `CFLAGS` that are set by MacPorts, which likely isn't what we want.

Reply:
upstream also states that it only works on 10.13.6 and above, which I can't really judge... Anyway, once this PR is finished and merged the buildbots will try and compile it on older systems as well. If it turns out that that indeed doesn't work then one would need to add some additional code to signify that it doesn't work on certain systems...

Reply:
I set the minimum xcode version based on the earliest one compatible with OSX 10.13.6 (see https://github.com/macports/macports-ports/pull/9006/files#diff-3dffc7e15962c8700de1f3577d0aa40aaa253065179a2c044576af59c9596b83R57). Is there an additional restriction I should add here? 

As per the c flags... I think I may just be misunderstanding something about the Macports ecosystem: it seems to me that the yabai author explicitly added those flags since they wanted the OSAX source files for be built as shared libraries, with position independent code support, etc. Are we meant to override that with whatever the user's CFLAGS are set to? Looking at my own Macports install, as well as our CI runs, it seems as though that would be a fairly major difference to how `yabai` gets built, which I imagine could be incompatible with assumptions the author made about their own build system (though I have not read through the full yabai codebase to be sure!).

Reply:
I am pretty sure that you should at least remove the `-O2 -mmacosx-version-min=10.13` and make sure that everywhere it uses the MacPorts flags to set the optimization level, the SDK and arch; you can certainly add whatever else upstream wants to add there I think. 

Admittedly, all these flags and Makefile business is still a bit black-magic to me and I typically try to follow what happens in other Portfiles or learn from mistakes I made earlier myself or comments on PRs that I merged in the past. Please try to accomplish what I suggested above and if the CI is happy and the correct flags are used then I am fairly confident in merging this. Of course if others comment beforehand here we should use their expertise, but otherwise someone will point out the mistakes after the fact - a good way of learning as well ;)

Reply:
Sure thing; I've amended the changeset to use $CFLAGS everywhere. It seems to all be working locally, and hopefully CI will work out as well. Please let me know if there are any other changes I should make!